### PR TITLE
Update ABSL to latest LTS version 20210324.2

### PR DESCRIPTION
**Pull request by yanavlasov**
Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

**Comment by moderation** [comment]
This PR may replace the abseil component of https://github.com/envoyproxy/envoy/pull/16293 - @kyessenov

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -105,12 +105,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Abseil",
         project_desc = "Open source collection of C++ libraries drawn from the most fundamental pieces of Google’s internal codebase",
         project_url = "https://abseil.io/",
-        version = "5d8fc9192245f0ea67094af57399d7931d6bd53f",
-        sha256 = "e3812f256dd7347a33bf9d93a950cf356c61c0596842ff07d8154cd415145d83",
+        version = "20210324.1",
+        sha256 = "441db7c09a0565376ecacf0085b2d4c2bbedde6115d7773551bc116212c2a8d6",
         strip_prefix = "abseil-cpp-{version}",
-        urls = ["https://github.com/abseil/abseil-cpp/archive/{version}.tar.gz"],
+        urls = ["https://github.com/abseil/abseil-cpp/archive/refs/tags/{version}.tar.gz"],
```
Comment (moderation) [inline bazel/repository_locations.bzl:111]: You need to revert this line back to what it was. Even though the GH release page has this URL it breaks some of our Python tooling scripts. I've confirmed that https://github.com/abseil/abseil-cpp/archive/20210324.1.tar.gz works and has the same `sha256`

**Review by moderation — CHANGES_REQUESTED** [review]

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -105,12 +105,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Abseil",
         project_desc = "Open source collection of C++ libraries drawn from the most fundamental pieces of Google’s internal codebase",
         project_url = "https://abseil.io/",
-        version = "5d8fc9192245f0ea67094af57399d7931d6bd53f",
-        sha256 = "e3812f256dd7347a33bf9d93a950cf356c61c0596842ff07d8154cd415145d83",
+        version = "20210324.1",
+        sha256 = "441db7c09a0565376ecacf0085b2d4c2bbedde6115d7773551bc116212c2a8d6",
```
Comment (moderation) [inline bazel/repository_locations.bzl:109]: Latest is now https://github.com/abseil/abseil-cpp/releases/tag/20210324.2
```
version = "20210324.2",
sha256 = "59b862f50e710277f8ede96f083a5bb8d7c9595376146838b9580be90374ee1f",
release_date = "2021-06-01",
```

**Review by moderation — CHANGES_REQUESTED** [review]

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (yanavlasov) [reply bazel/repository_locations.bzl]: done

File: bazel/repository_locations.bzl (hunk 2 above)
Comment (yanavlasov) [reply bazel/repository_locations.bzl]: done

**Review by wrowe — CHANGES_REQUESTED** [review]
This appears to be a breaking change on windows, please review and escalate for consideration...
'''
ERROR: C:/source/source/common/quic/BUILD:223:17: Compiling source/common/quic/envoy_quic_server_stream.cc failed: (Exit 2): cl.exe failed: error executing command 
  cd C:/build/tmp/execroot/envoy
  SET INCLUDE=C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.28.29910\include;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\shared;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\um;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\winrt;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\cppwinrt
    SET PATH=C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.28.29910\bin\HostX64\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\IDE\VC\VCPackages;C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\bin\Roslyn;C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\Tools\devinit;C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x64;C:\Program Files (x86)\Windows Kits\10\bin\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\\MSBuild\Current\Bin;C:\Windows\Microsoft.NET\Framework64\v4.0.30319;C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\IDE\;C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\Tools\;;C:\Windows\system32
    SET PWD=/proc/self/cwd
    SET TEMP=C:\Windows\Temp
    SET TMP=C:\Windows\Temp
    SET TMPDIR=C:\Windows\Temp
    SET USE_CLANG_CL=
  C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.28.29910/bin/HostX64/x64/cl.exe @bazel-out/x64_windows-opt/bin/source/common/quic/_objs/envoy_quic_server_session_lib/envoy_quic_server_stream.obj.params
Execution platform: @rbe_windows_msvc_cl//config:platform
external\com_google_absl\absl/types/compare.h(49): error C2338: Only literal `0` is allowed.
external\com_google_absl\absl/types/optional.h(711): note: see reference to class template instantiation 'absl::lts_20210324::compare_internal::Fail<T>' being compiled
        with
        [
            T=bool
        ]
'''

**Comment by yanavlasov** [comment]
We should fix the it here: https://github.com/envoyproxy/envoy/blob/b42563c8d951e59e66c4d0648ef9c5bfb67ffc2d/source/common/quic/envoy_quic_client_stream.cc#L316

I have pinged @danzh2010 about changing this as the absl::optional is extraneous.

**Comment by danzh2010** [comment]
> We should fix the it here:
> 
> https://github.com/envoyproxy/envoy/blob/b42563c8d951e59e66c4d0648ef9c5bfb67ffc2d/source/common/quic/envoy_quic_client_stream.cc#L316
> 
> I have pinged @danzh2010 about changing this as the absl::optional is extraneous.

eliminating absl::optional<bool> is not feasible here. Can you try using `has_value()` instead of comparison with absl::nullopt?

**Comment by yanavlasov** [comment]
Obsolete by PR #16687

**Pull request closed without merging**